File: app/src/main/java/com/dubedivine/samples/data/model/Answer.java
package com.dubedivine.samples.data.model;


import java.util.Date;
import java.util.List;

/**
 * Created by divine on 2017/08/13.
 */
public class Answer {
    private String body;
    private long votes;
    private boolean isChoosen;
    private Date createAt = new Date();
    private List<Comment> comments;
    private Media video;
    private long id = System.currentTimeMillis();

    public Answer() {
    }

    public Answer(String body, long votes, boolean isChoosen) {
        this.body = body;
        this.votes = votes;
        this.isChoosen = isChoosen;
    }

    public Answer(String body, long votes, boolean isChosen, List<Comment> comments, Media video) {
        this.body = body;
        this.votes = votes;
        this.isChoosen = isChosen;
        this.comments = comments;
        this.video = video;
    }

    public String getBody() {
        return body;
    }

    public long getVotes() {
        return votes;
    }

    public boolean isChoosen() {
        return isChoosen;
    }

    public Date getCreateAt() {
        return createAt;
    }

    public List<Comment> getComments() {
        return comments;
    }

    public Media getVideo() {
        return video;
    }

    public long getId() {
        return id;
    }
}

File: app/src/main/java/com/dubedivine/samples/data/model/Answer.kt
package com.dubedivine.samples.data.model


import java.io.Serializable
import java.util.*
import kotlin.collections.ArrayList

/**
 * Created by divine on 2017/08/13.
 */
class Answer : Serializable {
    var body: String? = null
        private set  // todo: this is ugly y not make it a val
    var votes: Long = 0
        private set
    var isChosen: Boolean = false
        private set
    val createAt = Date()
    var comments: ArrayList<Comment> = ArrayList()
    var video: Media? = null
    val files: List<Media>? = null
    val id: String? = null // UUID
    var user:User? = null

    constructor() {}

    constructor(body: String, votes: Long, isChosen: Boolean, user:User) {
        this.body = body
        this.votes = votes
        this.isChosen = isChosen
        this.user = user
    }

    constructor(body: String, votes: Long, isChosen: Boolean, comments: ArrayList<Comment>, video: Media, user:User) {
        this.body = body
        this.votes = votes
        this.isChosen = isChosen
        this.comments = comments
        this.video = video
        this.user = user
    }
}
